transcoder.go

```diff
@@ -1607,6 +1615,10 @@ func (w *transformingWriter) Write(data []byte) (n int, err error) {
 				w.rw.reportError(err)
 				return written, err
 			}
+			if w.trailerWritten && len(data) == 0 {
```

Comment:
Can the state flag `w.trailerWritten` be removed here and changed to:
```go
if w.latestEnvelope.trailer && len(data) == 0 { ... }
```

Reply:
Yeah, I think you're right. Done.